lightning/src/util/events.rs

```diff
@@ -108,6 +113,91 @@ pub enum Event {
 	},
 }
 
+impl Writeable for Event {
+	fn write<W: Writer>(&self, writer: &mut W) -> Result<(), ::std::io::Error> {
+		match self {
+			&Event::FundingGenerationReady { .. } => {
+				0u8.write(writer)?;
+				// We never write out FundingGenerationReady events as, upon disconnection, peers
+				// drop any channels which have not yet exchanged funding_signed.
+			},
+			&Event::FundingBroadcastSafe { ref funding_txo, ref user_channel_id } => {
+				1u8.write(writer)?;
+				funding_txo.write(writer)?;
+				user_channel_id.write(writer)?;
+			},
+			&Event::PaymentReceived { ref payment_hash, ref amt } => {
+				2u8.write(writer)?;
+				payment_hash.write(writer)?;
+				amt.write(writer)?;
+			},
+			&Event::PaymentSent { ref payment_preimage } => {
+				3u8.write(writer)?;
+				payment_preimage.write(writer)?;
+			},
+			&Event::PaymentFailed { ref payment_hash, ref rejected_by_dest,
+				#[cfg(test)]
+				ref error_code,
+			} => {
+				4u8.write(writer)?;
+				payment_hash.write(writer)?;
+				rejected_by_dest.write(writer)?;
+				#[cfg(test)]
+				error_code.write(writer)?;
+			},
+			&Event::PendingHTLCsForwardable { time_forwardable: _ } => {
+				5u8.write(writer)?;
+				// We don't write the time_fordwardable out at all, as we presume when the user
+				// deserializes us at least that much time has elapsed.
```

Comment:
About the assumption than some time has elapsed, I' m not sure, some people where interested by some HA scheme where you synchronize in real-time state between different ChannelManager. Dunno if it makes sense right now with current locking of ChannelHolder but still something to have in mind..

Reply:
Right, certainly isn't a thing you can do now, but also no big deal - if the user did some kind of crazy serialize-to-disk thing and then ends up relaying pending HTLCs a few seconds early, not really any harm done.